### DISCUSSION TITLE
SSH key erroneously unverified automatically

### DISCUSSION THREAD

**Discussion by iainhouston** [community/community#23195] [New to GitHub]
<p>Baffled by failure to push well-established repo to GitHub.<br>
GitHub reports</p>
<pre><code class="lang-auto">ERROR: We're doing an SSH key audit.
Reason: unverified automatically (private key found in a public repository)
Please visit https://github.com/settings/keys/&lt;a number&gt;
</code></pre>
<p>My <code>SSH and GPG keys</code> page shows no unverified keys and I have absolutely no hint as to where such a private key exists - and I am 99% sure there is no unencrypted key in this repo.</p>

**Comment by henwen** [comment]
<p>I’m having this problem as well and followed the instructions provided by GH <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/reviewing-your-ssh-keys" rel="noopener nofollow ugc">here</a> to no avail.</p>
<p>My SSH settings page doesn’t show an “Approve” button for my SSH key, and my key on GH matches up with what <code>ssh-add -l -E sha256</code> shows locally.</p>
<p>I temporarily solved the issue by deleting my key in GH, generating a new one, and adding it to GH. Unfortunately, when I tried a git command for the first time today, the issue came back.</p>
<p>I’m on a Mac running macOS Catalina 10.15.5.</p>
<p>Any help would be much appreciated!</p>

**Comment by skylerbala** [comment]
<p>Also experiencing the same issue</p>

**Comment by iainhouston** [comment, marked as answer]
<p>I’ve seemingly solved the issue here. Don’t know whether this was due to a mistake on my part or GitHub’s … I’m pretty sure that, on this occasion, the <em>cause</em> was my error even though GitHub reported the wrong error message and the wrong SSH key’s fingerprint in its error message.</p>
<p>I discovered my error by adding <code>IdentitiesOnly yes</code> to my <code>~/.ssh/config</code> for <code>Host GitHub.com</code></p>
<p>This revealed that I had mis-spelled the <code> IdentityFile ~/.ssh/github.rsa</code> line and the ssh key being used was not the one I intended.</p>
<p>So this is what I have now on my MacOS</p>
<pre><code class="lang-auto">Host *
     IdentityFile ~/.ssh/id_rsa
     UseKeychain yes
     ForwardX11 no
     AddKeysToAgent yes
     ForwardAgent yes

Host github.com
     User git
     Hostname github.com
     PreferredAuthentications publickey
     IdentityFile ~/.ssh/github.rsa
     IdentitiesOnly yes
</code></pre>

**Comment by henwen** [comment]
<p>You may be onto something <a class="mention" href="/iainhouston">@iainhouston</a>.</p>
<p>I just had the wildcard Host entry in my ssh config. Added an explicit entry for <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> like you have and it’s working now!</p>

**Reply by sandeepV2** [reply]
Thanks @iainhouston and @henwen.  adding ssh config for git hub did work.

**Comment by iainhouston** [comment]
<p>Brilliant! Glad it’s working for you.</p>
<p>I still think there’s a bug in that GitHub is reporting a misleading error message and the wrong fingerprint.</p>
<p>Will look into how to report bugs to GitHub.</p>

**Comment by iainhouston** [comment]
<p><a class="mention" href="/henwen">@henwen</a> I reported this issue at <a href="https://support.github.com/contact/feedback" rel="noopener nofollow ugc">https://support.github.com/contact/feedback</a></p>

**Comment by fantasticrice** [comment]
<p>I only have the wildcard entry, but your answer helped me solve the problem. I stopped receiving the error when I added these config settings that I copied from your above solution:</p>
<pre><code class="lang-auto">PreferredAuthentications publickey
IdentitiesOnly yes
</code></pre>

**Comment by svadr** [comment]
<p>Thank you so so much, this just happened me today and adding</p>
<pre><code class="lang-auto">  PreferredAuthentications publickey
  IdentitiesOnly yes
</code></pre>
<p>solved my issue.</p>

**Comment by alozie** [comment]
<p>I concur. This issue’s been plaguing me for the past month! The fix from post <a href="https://github.community/t/ssh-key-erroneously-unverified-automatically/141837/8">#8</a> (adapted from post <a href="https://github.community/t/ssh-key-erroneously-unverified-automatically/141837/4">#4</a>) worked for me as well.</p>
<p>Thank you all!!!</p>

**Comment by vasanthsadasivan** [comment]
Had this same issue a few times within two days lost like 4 hrs finding solution or work around. I don't think its solely gh issue because when I ran ssh-add -l -E sha256, there were no identities listed even when there were identities listed prior. This might've occurred after I closed a few shells in the term but read on to see how it could also be a gh issue also . I think its a combo of both.

If there is no identities listed ssh-add -l -E sha256, 
add private keys (same file name without the .pub extension): 
command: (check https://linux.die.net/man/1/ssh-add)
ssh-add .ssh-dir-path/id_ed25519        
( this will ask for paraphrase (password), if u forgot it delete .ssh-dir-path/id_ed25519 file and regenerate a new one and u can save to keychain (check https://linux.die.net/man/1/ssh-keygen)
command: (check https://linux.die.net/man/1/ssh-add)
ssh-add .ssh-dir-path/id_rsa

add to file ssh-dir-path/known_hosts
github.com ssh-ed25519
"your public key" (key in id_ed25519 .pub)
github.com ssh-rsa
"your public key" (key in id_rsa.pub)

Then make sure your key for gh matches by running the command 'gh ssh-key list' this is same  as https://github.com/settings/keys once logged in, u will not see the actual public key listed there only its name if u give it one with the -t flag (read below) under ssh keys. What u do see as a sort key is the fingerprint (i think). Now here's where the funny stuff happens, so I deleted the key using 'gh ssh-key delete {id#}' which is listed when u run 'gh ssh-key list',then verify I am not being authenticated during a git push with same error and sure enuff I get error:  
ERROR: We're doing an SSH key audit.
Reason: unverified automatically (private key found in a public repository)
Please visit https://github.com/settings/keys/27394849 to approve this key so we know it's safe.
Fingerprint:
SHA256:************************************
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists. 

So I re-add the key with 'gh ssh-key add  {ssh-dir-path}/id_ed25519.pub -t "{name:string}", then all works.
